### PR TITLE
generic cli

### PR DESCRIPTION
**Pull request by 4meta5**
following sunshine-identity norms

### CODE REVIEW

File: pallets/donate/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+#[test]
+fn check_donation_accuracy() {
+    new_test_ext().execute_with(|| {
+        let one = Origin::signed(1);
+        let two = Origin::signed(2);
+        assert_eq!(Balances::total_balance(&2), 98);
+        assert_ok!(
+            Donate::make_donation_in_proportion_to_ownership_without_fee(
+                one, 1, 60
+            )
```
Comment (dvc94ch) [inline pallets/donate/src/tests.rs:170]: Maybe call it make_prop_donation

Comment (4meta5) [reply pallets/donate/src/tests.rs:170]: I like long variable names. It's my style bro

**Comment by 4meta5** [comment]
**Note to self**, add docs for how the string needs to be input for `BatchIssueShares`, `BatchBurnShares`, and `RegisterWeightedOrg`

This implementation is in `client/clientsrc/utils` and it is used in `client/cli/` which is used in the `test-cli`

```rust
#[derive(Clone, Debug)]
pub struct AccountShare(pub String, pub u64);
impl FromStr for AccountShare {
    type Err = Error;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        let coords: Vec<&str> = s.trim_matches(|p| p == '(' || p == ')' )
                                 .split(',')
                                 .collect();
        let acc_str = coords[0];
        let share_fromstr = coords[1].parse::<u64>().map_err(|_| Error::ParseIntError)?;
        Ok(AccountShare(acc_str.to_string(), share_fromstr))
    }
}
```

**Comment by 4meta5** [comment]
TODO

- [ ] add permissions trait to bank2 to separate it from underlying logic
- [ ] finish bank2 with tests
- [ ] update bounty to use bank2 instead of bank; remove bank and replace `pallet-utils/src/bank.rs` with `bank2/src/bank.rs`
- [ ] add bank logic to cli
- [ ] add bounty logic to cli

remove CalculateOwnership trait from org, that logic will only be in `donate` because it's still lossy af

Logistics
- [ ] update readme
- [ ] 0.0.6 release?

**Comment by 4meta5** [comment]
more TODO

- [ ] add spend functionality natively to bank by exercising `vote` functionality, already inherited

File: pallets/donate/src/tests.rs (hunk 1 above)
Comment (4meta5) [reply pallets/donate/src/tests.rs:170]: Oh, because the macro. Ye, I'll change it....

**Comment by 4meta5** [comment]
I'll finish these TODOs and then merge. The last few things are:
- [ ] add vote logic to CLI
- [ ] add bounty logic to CLI
- [ ] 0.0.6 release
- [ ] update readme

This will finish up everything necessary for the demo

**Comment by 4meta5** [comment]
I don't know why I'm getting this error but I suspect that it's because the `bank::Bank` trait does not inherit `donate::Donate` but it does inherit it in the actual module. Very misleading error message if this suspicion is proven correct.

```rust
➜  client git:(generic-cli) ✗ carb
   Compiling sunshine-bounty-client v0.1.0 (/Users/4meta5/sunshine-protocol/sunshine-node/client/client)
warning: unused import: `bounty::*`
 --> client/client/src/abstract.rs:7:9
  |
7 |         bounty::*,
  |         ^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0599]: no method named `org_bank_account_opened` found for associated type `<T as substrate_subxt::system::System>::Hash` in the current scope
   --> client/client/src/client.rs:349:14
    |
349 |             .org_bank_account_opened()
    |              ^^^^^^^^^^^^^^^^^^^^^^^ method not found in `<T as substrate_subxt::system::System>::Hash`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `srml::bank::OrgBankAccountOpenedEventExt` defines an item `org_bank_account_opened`, perhaps you need to implement it
   --> client/client/src/srml/bank.rs:95:39
    |
95  | #[derive(Clone, Debug, Eq, PartialEq, Event, Decode)]
    |                                       ^^^^^
    = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error; 1 warning emitted
```

**Comment by 4meta5** [comment]
I'm just going to add a CHANGELOG and then merge.

**Pull request merged**